java.lang.IllegalArgumentException: CARROTS isn't an item

### What steps will reproduce the problem?

```
com.destroystokyo.paper.exception.ServerEventException: Could not pass event InventoryClickEvent to Towny v0.100.3.12
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:72)
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:131)
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628)
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClick(ServerGamePacketListenerImpl.java:3217)
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:69)
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:33)
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:56)
	at net.minecraft.server.TickTask.run(TickTask.java:18)
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:151)
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24)
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1545)
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:201)
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:125)
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1522)
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1515)
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:135)
	at net.minecraft.server.MinecraftServer.managedBlock(MinecraftServer.java:1474)
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1481)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1326)
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:331)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: CARROTS isn't an item
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:143)
	at org.bukkit.inventory.ItemStack.of(ItemStack.java:58)
	at org.bukkit.inventory.ItemStack.<init>(ItemStack.java:138)
	at org.bukkit.inventory.ItemStack.<init>(ItemStack.java:117)
	at org.bukkit.inventory.ItemStack.<init>(ItemStack.java:104)
	at org.bukkit.inventory.ItemStack.<init>(ItemStack.java:87)
	at Towny.jar//com.palmergames.bukkit.towny.utils.ResidentUtil.openGUIInventory(ResidentUtil.java:141)
	at Towny.jar//com.palmergames.bukkit.towny.listeners.TownyInventoryListener.onClick(TownyInventoryListener.java:129)
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor567.execute(Unknown Source)
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77)
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
	... 20 more
```

### What is the expected output?

Not sure, I'm pretty sure we didn't add manually carrots anywhere in Towny config, but we did upgrade map and plugin a few times in between without resetting config (from 1.20.6 and Towny 100ish? to current versions)

### Towny version

Towny v0.100.3.12

### Server version

Paper 1.21.1 https://github.com/PaperMC/paper/commit/e08e6679fcaf5ce8b91db628309ed530e58a4133

### Please use Pastebin.com to link the following files

1. Your full server startup from the logs\latest.log:
2. Your Towny config.yml:
3. Your townyperms.yml:
4. Your log's error:

(optional) If this is to do with permissions,
5. Your permissions file:

(optional) If this is to do with chat,
6. Your chatconfig.yml:
7. Your channels.yml:

--- a/Towny/src/main/java/com/palmergames/bukkit/towny/utils/ResidentUtil.java
+++ b/Towny/src/main/java/com/palmergames/bukkit/towny/utils/ResidentUtil.java
@@ -43,6 +43,7 @@ import com.palmergames.bukkit.towny.object.Town;
 import com.palmergames.bukkit.towny.object.TownyInventory;
 import com.palmergames.bukkit.util.BukkitTools;
 import com.palmergames.bukkit.util.Colors;
+import com.palmergames.bukkit.util.MaterialUtil;
 import com.palmergames.util.TimeMgmt;
 
 import net.kyori.adventure.text.Component;
@@ -138,8 +139,11 @@ public class ResidentUtil {
 	public static void openGUIInventory(Resident resident, Collection<Material> set, String name) {
 		ArrayList<ItemStack> items = new ArrayList<>();
 		for (Material material : set) {
-			if (!material.isItem())
-				continue;
+			if (!material.isItem()) {
+				material = MaterialUtil.getItemFromMaterial(material);
+				if (material == null)
+					continue;
+			}
 			
 			items.add(new ItemStack(material));
 		}

--- a/Towny/src/main/java/com/palmergames/bukkit/towny/utils/ResidentUtil.java
+++ b/Towny/src/main/java/com/palmergames/bukkit/towny/utils/ResidentUtil.java
@@ -137,8 +137,12 @@ public class ResidentUtil {
 	
 	public static void openGUIInventory(Resident resident, Collection<Material> set, String name) {
 		ArrayList<ItemStack> items = new ArrayList<>();
-		for (Material material : set)
+		for (Material material : set) {
+			if (!material.isItem())
+				continue;
+			
 			items.add(new ItemStack(material));
+		}
 		
 		createTownyGUI(resident, items, name);
 	}

